File: Day2_ControlFlow_Go/easy/easy_03.go
// FILE: easy_03.go
// TOPIC: Control Flow Deep Dive - Day 2 (Go)
// DIFFICULTY: Easy
//
// TASK: While and Do-While loops in Go (using the `for` keyword)
// --------------------------------------------------------------
// Go has NO `while` or `do while` keyword! Everything is a `for` loop:
//   - C++ while (cond)   -> Go: for cond {}
//   - C++ do-while       -> Go: for { ... if !cond { break } }
//
// 1. Declare: count := 10
// 2. Write a while-style loop in Go (`for count < 5`) that prints:
//      "While running: " + count
// 3. Write a do-while style loop in Go that is guaranteed to run at least once:
//      for {
//          // body
//          if !(count < 5) { break }
//      }
//    Printing: "Do-While running: " + count
//
// EXPECTED OUTPUT:
//   Do-While running: 10
//   Explanation: do-while executes the body at least once before checking the condition.

package main

import "fmt"

func main() {
	// TODO: Write your solution here.

}

File: Day2_ControlFlow/easy/easy_03.cpp
/*
 * FILE: easy_03.cpp
 * TOPIC: Control Flow Deep Dive - Day 2
 * DIFFICULTY: Easy
 *
 * TASK: while vs do-while Loop Guarantee
 * ----------------------------------------
 * A `while` loop checks the condition BEFORE running the body (may run 0 times).
 * A `do while` loop runs the body FIRST, then checks the condition (guaranteed >= 1 run).
 *
 * 1. Declare: int count = 10;
 * 2. Write a `while (count < 5)` loop that prints "While running: " << count.
 * 3. Write a `do while (count < 5)` loop that prints "Do-While running: " << count.
 * 4. Observe the output: which loop executed and which one was skipped?
 * 5. Print a concluding message explaining why the do-while executed even though
 *    10 is NOT less than 5.
 *
 * EXPECTED OUTPUT:
 *   Do-While running: 10
 *   Explanation: do-while executes the body at least once before checking the condition.
 */

#include <iostream>
using namespace std;

int main() {
    // TODO: Write your solution here.
    int count = 10;

    while (count < 5) {
        std::cout << "While running : " << count << "\n";
    };

    do {
        std::cout << "Do-While running: " << count << "\n";
    } while (count < 5);

    return 0;
}
